pages/recipes/[slug].js

```diff
@@ -16,18 +16,14 @@ export default function OneRecipe({ recipe }) {
 // STEP 1: tell next.js how many recipe pages it needs to make
 export async function getStaticPaths() {
   // go get all recipes
-  const recipes = await sanityClient.fetch(
-    `*[_type == "recipe"]{
-      name,
-      slug
-    }`
+  const paths = await sanityClient.fetch(
+    `*[_type == "recipe" && defined(slug.current)]{
```

Comment:
`defined(slug.current)` makes sure that we only get recipes that have a slug. We also just make the required data structure in GROQ so that we don't need to do any extra lifting with JS.